rpg/button.py

```diff
@@ -0,0 +1,108 @@
+import pygame
+from typing import Union, Optional, List, Tuple, Callable
+
+
+def load_surface(surface_ref: Union[str, pygame.Surface, Tuple[int, int, int]]) -> Optional[pygame.Surface]:
+    """Load a pygame surface
+
+    Args:
+        surface_ref: the file path to the surface OR a pygame surface OR a rgb color
+
+    Returns:
+        the loaded surface, None if fails
+    """
+    try:
+        if type(surface_ref) == str:
+            return pygame.image.load(surface_ref)
+        elif type(surface_ref) == pygame.Surface:
+            return pygame.Surface
+        elif type(surface_ref) in [tuple, list]:
+            surface = pygame.Surface((1, 1))
+            pygame.draw.rect(surface, surface_ref, [0, 0, 1, 1])
+            return surface
+        else:
+            return None
+    except:
+        return None
+
+
+class Button:
+    pos_x: float
+    pos_y: float
+    width: float
+    height: float
+    image_default: pygame.Surface
+    image_click: pygame.Surface
+    _image: pygame.Surface
+    onclicks: List[Callable]
+
+    def __init__(self, x: float, y: float, w: float, h: float,
+                 image_default=None,
+                 image_hover=None,
+                 image_click=None):

```

Comment:
Make `image_default` a required argument. Remove the `w` and `h` and derive them.